vali_objects/position.py

```diff
@@ -574,3 +581,8 @@ def _update_position(self):
             # If the position is already closed, we don't need to process any more orders. break in case there are more orders.
             if self.position_type == OrderType.FLAT:
                 break
+
+        # calculate_drawdown
+        self.max_return = max(self.max_return, self.return_at_close)
+        self.drawdown = 1.0 + ((self.return_at_close - self.max_return) / self.max_return)
+        self.mdd = min(self.mdd, self.drawdown)
```

Comment:
this code should go in the set_returns function so these values get updated in mdd_checker periodically.